hibernate-core/src/main/java/org/hibernate/sql/exec/internal/JdbcSelectExecutorStandardImpl.java

```diff
@@ -158,7 +158,7 @@ public String getEffectiveOptionalEntityName() {
 
 			@Override
 			public Object getEffectiveOptionalId() {
-				return null;
+				return executionContext.getEntityId();
```

Comment:
We seems to need the non-null id value here so the only usage of 'getEffectiveOptionalInstance()' in `AbstractEntityInitilizer` (see below) could work because of the id comparison precondition